src/walk_dirs.rs

```diff
@@ -118,8 +123,12 @@ pub fn discover_fences_and_files(start_path: &str) -> Vec<WalkFileData> {
                                     || file_name.ends_with(".jsx")
                                     || file_name.ends_with(".js")
                                 {
-                                    let file_path = dir_entry.parent_path.join(file_name);
+                                    let mut file_path = dir_entry.parent_path.join(file_name);
+                                    #[cfg(target_os = "windows")] {
```

Comment:
Suggestions for this are welcome 😬

Reply:
This gets used twice, and I'm not a huge fan of the only sometimes mutable file_path. I believe it will warn when compiled on osx/linux. It would also be good to handle the possible error here.

```rust
// consume a path and produce a relative path
fn as_relative_path(p: Path) -> Result<&RelativePath, Error> {
  let slashed: PathBuf
  #[cfg(target_os = "windows")] {
    slashed = match file_path.to_slash() {
      Ok(slash_path) => PathBuf::from(slash_path.to_string()),
      // you would need to figure out how best to define this error type,
      // possibly by adding some new WalkDirsError enum
      Err => { return Err(...) }
    }
  }
  #[cfg(not(target_os = "windows"))] {
    slashed = PathBuf::from(p)
  }
  return RelativePath::from_path(&file_path)
}
```